nautobot/core/views/mixins.py

```diff
@@ -473,7 +474,14 @@ def get_queryset(self):
         Override the original `get_queryset()` to apply permission specific to the user and action.
         """
         queryset = super().get_queryset()
-        return queryset.restrict(self.request.user, PERMISSIONS_ACTION_MAP[self.action])
+        action_map = self.get_action_map()
+        return queryset.restrict(self.request.user, action_map[self.action])
+
+    def get_action_map(self):
+        """Provides the ability to define which permission is needed for a custom action."""
+        if self.custom_action_permission_map:
+            return {**PERMISSIONS_ACTION_MAP, **self.custom_action_permission_map}
+        return PERMISSIONS_ACTION_MAP
```

Comment:
Instead of having to define a custom actions in a mapping set as a class variable, we should just have it return the name of the action if it's not found.

I would propose a `get_action()` method that does something like this:

```python
    def get_action(self):
       return PERMISSIONS_ACTION_MAP.get(self.action, self.action)
```

And then the places that need to access the action:

```python
    return queryset.restrict(self.request.user, self.get_action())
```

Also you appear to have missed another reference to `PERMISSIONS_ACTION_MAP`  inside of `get_queryset()`.

Reply:
I like that! And just document you'd need the custom action in the permissionsset

Reply:
This doesn't provide the ability for a custom action to use one of the default permissions though, does it? It *requires* that a custom action have an associated custom permissions?